pkg/eventloop/eventloop_internal_test.go

```diff
@@ -0,0 +1,55 @@
+package eventloop
+
+import (
+	"reflect"
+	"sync"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestLoopRun(t *testing.T) {
```

Comment:
I would also add a test with multiple go routines, to test thread-safety

Reply:
@coderabbitai Generate unit tests for eventloop that tests thread-safety

Reply:
@coderabbitai The Remove method should be thread-safe as well, add tests for it too

Reply:
I added tests but your suggestion doesn't match the code of the implemented event loop.